influunt-app/app/plugins/influunt/visualizacao-simulador.js

```diff
@@ -155,6 +155,35 @@ var influunt;
           }
         }
 
+        function removeFuture(next){
+          var removeAfter, i;
+          removeAfter = next ? (((parseInt(tempo / 256)+1) * 2560) + MARGEM_LATERAL) :
+                               ((parseInt(tempo / 256) * 2560) + MARGEM_LATERAL);
+
+          for(i = intervalosGroup.children.length - 1; i >= 0; i--) {
+            if(intervalosGroup.children[i].x >= removeAfter){
+              intervalosGroup.children[i].body = null;
+              intervalosGroup.children[i].destroy();
+            }
+          }
+
+          for(i = eventosGroup.children.length - 1; i >= 0; i--) {
+            if(eventosGroup.children[i].x >= removeAfter){
+              eventosGroup.children[i].body = null;
+              eventosGroup.children[i].destroy();
+            }
+          }
```

Comment:
Para diminuir um pouco de duplicação aqui, não ficaria melhor criar um metodo apenas com um for, passando apenas o intervalosGroup ou eventosGroup ?